patterns/README.md

```diff
@@ -1030,3 +1030,371 @@ A field MAY have a `missingValues` property that MUST be an `array` where each e
+  "missingValues": ["."]
+}
+```
+
+This represents encoded data exported from software with support for  value
+labels. The values `.a`, `.b`, etc. are known as *extended missing values*
+(Stata and SAS only) and provide 26 unique missing values for numeric fields
+(both integer and float) in addition to the system missing value ("`.`"); in
+SPSS these would be replaced with designated numbers (e.g., -97, -98 and -99).
+
+Note that one might argue that the encoding property should instead be
+specified as:
+
+```
+{
+  "encoding": {
+    "Poor": 1,
+    "Fair": 2,
+    "Good": 3,
+    "Very good": 4,
+    "Excellent": 5
+}
+```
+
+since that represents the encoding that has been applied to the data, and the
+table in the example is what is now necessary to *decode* the data. However,
+there are at least three arguments in favor of the proposed specification.
+First, it is the way value labels are uniformly written (e.g., in Stata, SAS
+and SPSS). Second, it automatically imposes the necessary constraint that the
+codes are unique (since a JSON object's keys must be unique). Third, it
+simplifies working with the encoding programmatically, since it can be read as
+an associative array and then applied directly to decode to the data (e.g.,
+using `DataFrame.replace()` in Pandas).
+
+### Specification
+
+1. A field MAY have a `missingValues` property that MUST be an `array` where
+   each entry is a `string`. If not specified, each field shall inherit the
+   entries in the `missingValues` property at the level of the tabular data
+   resource. If present at both the field and resource levels, the
+   field level property will be replaced by the *union* of the two arrays,
+   with the values specified at the resource level appearing in the same order
+   *after* those specified at the field level.
+
+2. A field with an `enum` constraint or an `encoding` property MAY have an
+   `ordered` property that MUST be a boolean. A value of `true` indicates that
+   the field should be treated as having an ordinal scale of measurement, with
+   the ordering given by the order of the field's `enum` array or by the
+   lexical order of the `encoding` object's keys, with the latter taking
+   precedence. Fields without an `enum` constraint or an `encoding` property
+   or for which the encoding object's keys do not include all values observed
+   in the data (excluding any values specified in either the field level or
+   resource level `missingValues` property) SHOULD NOT have an `ordered`
+   property since in that case the correct ordering of the data is ambiguous.
+   The absence of an `ordered` property MUST NOT be taken to imply
+   `ordered: false`.
+
+3. A field MAY have an `encoding` property that MUST be an object. This
+   property SHOULD be used to indicate how the values in the data (represented
+   by the object's keys) are to be labeled or translated (represented by the
+   corresponding value). The object's keys MAY include values that do not
```

Comment:
Should the requirement for the keys to be listed as strings in `encoding` be mentioned here?

Reply:
Good point; I'll add this.

Reply:
Suggested change made in d03ede0c869bb69f55f2d7a21bdd373949f0ae90.